### ISSUE TITLE
Relax `pytz` dependency constraint

### ISSUE THREAD

**Issue by MicahLyle** [uploadcare/pyuploadcare#206]
### Describe the bug

This isn't a bug per-say, but `pytz` is pinned to `< 2022`. This means that I can't use `pyuploadcare` alongside my code library if I want `pytz >= 2022`.

```
The conflict is caused by:
    The user requested pytz==2022.2
    pyuploadcare 3.0.0 depends on pytz<2022.0 and >=2021.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

### Expected behavior

The `pyuploadcare` library should work with later `pytz` versions as well, so that we can have up to date timezone definitions even if `pyuploadcare` hasn't needed a release in a while.

**Comment by MicahLyle** [comment]
Would also be nice if `typing-extensions` version could be bumped at the next release.

**Comment by dmitry-mukhin** [comment]
closed via #210
